src/rdf4cpp/rdf/datatypes/registry/DatatypeRegistry.hpp

```diff
@@ -48,6 +49,35 @@ class DatatypeRegistry {
         unop_fptr_t neg_fptr;  // -a
     };
 
+    struct NumericOpsStub {
+        /**
+         * Determines the number of subtype levels to immediately skip up in
+         * the hierarchy, if searching for a _numeric_ type conversion.
+         *
+         * @example if start_s_off = 1 then the search will _start_ at the immediate supertype
+         *      of the type this NumericOpsStub belongs to.
+         */
+        size_t start_s_off;
+    };
+
+    struct NumericOps : std::variant<NumericOpsStub, NumericOpsImpl> {
```

Comment:
```suggestion
    struct NumericOps : public std::variant<NumericOpsStub, NumericOpsImpl> {
```

It's just for readability. But I had to look up if parent classes of structs are public by default.